jax/_src/interpreters/pxla.py

```diff
@@ -1353,21 +1356,28 @@ def _hlo_unshard(ctx: mlir.LoweringRuleContext, aval, axis_env, out_axis, xs, pl
       perm.insert(out_axis, 0)
       transposed_dims = list(dims)
       transposed_dims.insert(out_axis, axis_env.sizes[-1])
-      aval = aval.update(shape=transposed_dims)
       out = hlo.TransposeOp(out, mlir.dense_int_elements(perm)).result
 
-    # TODO(mattjj): remove this logic when AllReduce PRED supported on CPU / GPU
     if convert_bool:
       float_zero = mlir.full_like_aval(ctx, 0, padded_aval)
       out = hlo.CompareOp(
           out,
           float_zero,
           hlo.ComparisonDirectionAttr.get("NE"),
           compare_type=hlo.ComparisonTypeAttr.get("FLOAT")).result
-    return out
+    return [out]
+
+  # TODO(mattjj): remove this logic when AllReduce PRED supported on CPU / GPU
+  if np.issubdtype(aval.dtype, np.bool_):
+    return mlir.lower_multi_platform(
```

Comment:
We should never call into multi-platform lowering from regular lowering, surely?

Reply:
This code is used for both single- and multi-platform lowering. I think it is Ok to call into lower_multi_platform, because that function falls back to the same code path as single-platform lowering if we are in single-platform lowering, or if there is a single lowering rule for all the current lowering platforms.

The alternative would be to add a conditional here. Do you think that is preferable? Perhaps it is, for as long as we don't yet trust the multi-platform lowering. wdyt?

Reply:
This is not necessary anymore, we have cleaned up the platform dependency for the hlo_unshard in a separate change.